corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +147,117 @@ def data_dictionary_json(request, domain, case_type_name=None):
+            name=case_type_name
+        )
+        case_type_data = _get_case_data(case_type)
+
+        properties_queryset = CaseProperty.objects.select_related('group').filter(case_type=case_type)
+        properties_queryset = properties_queryset.order_by('group_id', 'index', 'pk')[skip:skip + limit]
+        properties_queryset = properties_queryset.prefetch_related(
+            Prefetch('allowed_values', queryset=CasePropertyAllowedValue.objects.order_by('allowed_value'))
+        )
+
+        current_url = request.build_absolute_uri()
+        links = {"self": update_url_query_params(current_url, {"skip": skip, "limit": limit})}
+        if skip:
+            links["previous"] = update_url_query_params(
+                current_url,
+                {"skip": max(skip - limit, 0), "limit": limit}
+            )
+        if case_type_data["properties_count"] > (skip + limit):
+            links["next"] = update_url_query_params(current_url, {"skip": skip + limit, "limit": limit})
+        case_type_data["_links"] = links
+
+        case_type_data["groups"] = []
+        data_validation_enabled = toggles.CASE_IMPORT_DATA_DICTIONARY_VALIDATION.enabled(domain)
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        for group_id, props in itertools.groupby(properties_queryset, key=attrgetter("group_id")):
+            props = list(props)
+            grouped_properties = []
+            for prop in props:
+                prop_data = {
+                    'id': prop.id,
+                    'description': prop.description,
+                    'label': prop.label,
+                    'fhir_resource_prop_path': fhir_resource_prop_by_case_prop.get(prop),
+                    'name': prop.name,
+                    'deprecated': prop.deprecated,
+                    'is_safe_to_delete': prop.name not in used_props and prop.name != geo_case_prop,
+                    'index': prop.index,
+                }
+                if data_validation_enabled:
+                    prop_data.update({
+                        'data_type': prop.data_type,
+                        'allowed_values': {
+                            av.allowed_value: av.description
+                            for av in prop.allowed_values.all()
+                        },
+                    })
+                grouped_properties.append(prop_data)
+
+            group_data = {
+                "name": "",
+                "properties": grouped_properties
+            }
+            # Note that properties can be without group
+            if group_id:
+                group = props[0].group
+                group_data.update({
+                    "id": group.id,
+                    "name": group.name,
+                    "description": group.description,
+                    "deprecated": group.deprecated,
+                })
+            case_type_data["groups"].append(group_data)
+
+        return JsonResponse(case_type_data)
```

Comment:
Should the `geo_case_property` not also be returned here? After the case property data has been loaded we use the property to identify which ones are used in the Geospatial feature.

Reply:
Good question. Although not very sure on that, I think this was probably not required as Norman has completed the changes on the JS side.